lib/widgets/payment_status_sheets/processing_payment_sheet.dart

```diff
@@ -54,70 +61,141 @@ class ProcessingPaymentSheetState extends State<ProcessingPaymentSheet> {
   @override
   void initState() {
     super.initState();
-    _payAndClose();
+    _processPaymentAndClose();
   }
 
-  void _payAndClose() {
-    widget.paymentFunc().then((dynamic payResult) async {
+  void _processPaymentAndClose() {
+    widget.paymentFunc().then(_handlePaymentSuccess).catchError(_handlePaymentError);
+  }
+
+  void _handlePaymentSuccess(dynamic payResult) {
+    if (!mounted) {
+      return;
+    }
+
+    if (widget.isLnPayment) {
+      _handleLnPaymentResult(payResult);
+    } else if (widget.isLnUrlPayment) {
+      _handleLnUrlPaymentResult(payResult);
+    } else {
+      _closeSheetOnCompletion();
+    }
+  }
+
+  void _handleLnPaymentResult(dynamic payResult) {
+    if (payResult is SendPaymentResponse) {
+      _trackLnPaymentEvents(payResult);
+    } else {
+      _onPaymentFailure();
+    }
+  }
+
+  void _handleLnUrlPaymentResult(dynamic payResult) {
+    if (payResult is LnUrlPayResult_EndpointSuccess) {
+      _showSuccessAndClose(payResult);
+    } else {
+      _onPaymentFailure();
+    }
+  }
+
+  void _trackLnPaymentEvents(SendPaymentResponse payResult) {
+    final InputCubit inputCubit = context.read<InputCubit>();
+
+    final Future<void> paymentSuccessFuture = inputCubit.trackPaymentEvents(
+      payResult.payment.destination,
+      paymentType: PaymentType.send,
+    );
+    final Future<void> timeoutFuture = Future<void>.delayed(const Duration(seconds: 10));
```

Comment:
Perhaps we can use the timeout function on future? https://api.flutter.dev/flutter/dart-async/Future/timeout.html